### DISCUSSION TITLE
I can't download the package typesetting

### DISCUSSION THREAD

**Discussion by SpectatorLife** [fyne-io/fyne#4769] [Q&A] [answered]
golang for windows
`go get fyne.io/fyne/v2 `- ok
Next I'm trying to run an example
`go run hello.go`
error: cannot find package "github.com/go-text/typesetting/opentype/api"

**Comment by andydotxyz** [comment]
Maybe "go mod tidy" will fix the missing dependencies. What version of Go?

**Comment by SpectatorLife** [comment]
go version
`go version go1.21.6 windows/amd64`
go mod tidy
`go: modules disabled by GO111MODULE=off; see 'go help modules'`
go env -w GO111MODULE=on
`warning: go env -w GO111MODULE=... does not override conflicting OS environment variable`

**Reply by Jacalz** [reply, marked as answer]
I think you should try to get your modules to work instead of disabling them. Module support is the recommended way for dependencies

**Comment by SpectatorLife** [comment]
thanks, I'll try to overcome it
`go env -w GO111MODULE=... does not override conflicting OS environment variable`
I can't decide right away

**Comment by andydotxyz** [comment]
It seems like somewhere in your environment you have set `GO111MODULE=off` - Go no longer supports this configuration.

**Comment by SpectatorLife** [comment]
It looks like it, but the problem is that now I can’t reset it ((
 `go env -w GO111MODULE=... does not override conflicting OS environment variable`

**Reply by andydotxyz** [reply]
You'll need to figure out where in your environment it has been set and unset it. This seems to be a user or system configuration problem.

**Comment by SpectatorLife** [comment]
`does not override conflicting OS environment variable`
I'm the one who won. set the variable. Now after `go mod init` and `go mod tidy`  `go build` displays: ```no required module provides package fyne.io/fyne/v2/app: go.mod file not found in current directory or any parent directory```
although the files have downloaded in c:\Go\fyne.io\fyne\v2@v2.4.4\ 
`set GOPATH=C:\go`

**Reply by andydotxyz** [reply]
> although the files have downloaded in c:\Go\fyne.io\fyne\v2@v2.4.4

That's not the location that modules would place the code.

> go.mod file not found in current directory

It sounds like you may have changed directory between "go mod init" and "go build", otherwise it would find the file.

**Comment by SpectatorLife** [comment]
Thank you. deleted and re-initialized init and tidy. installed gcc and the example started)) thanks
